app/cdap/components/Replicator/Create/Content/Advanced/index.tsx

```diff
@@ -141,15 +141,26 @@ const AdvancedView: React.FC<ICreateContext & WithStyles<typeof styles>> = ({
   numInstances,
   setAdvanced,
 }) => {
-  const [localNumInstances, setLocalNumInstances] = React.useState(numInstances || 1);
-  const [taskSelection, setTaskSelection] = React.useState(TASK_OPTIONS.calculate);
-  const [dataAmount, setDataAmount] = React.useState(1);
+  const [localNumInstances, setLocalNumInstances] = useState(numInstances || 1);
+  const [taskSelection, setTaskSelection] = useState(TASK_OPTIONS.calculate);
+  const [dataAmount, setDataAmount] = useState(1);
+  const saveRef = useRef(localNumInstances);
```

Comment:
why is useRef necessary here? isn't useRef usually used to reference an element?

Reply:
useEffect cleanup functions don't have access to the most recent state

Reply:
to clarify - with an empty dependency array and mimicking componentWillUnmount it doesn't have access to the newest state

Reply:
I feel like useRef is generally used for DOM element reference. Perhaps what you want is useCallback instead?

Reply:
@elfenheart in order to get access to the most recent state in the "unmount" callback you need to make a reference to that value in the state. Its similar to this pr that @lahwran made here https://github.com/cdapio/cdap-ui/pull/92/files - its a pretty widely used pattern with examples in other places online (that I can find if necessary) - I can update this code to use @lahwran 's unmount callback if needed but that code was written after this pr.